qdrant-landing/content/articles/multitenancy.md

```diff
@@ -0,0 +1,189 @@
+
+Let's continue and start adding data. As you upsert your vectors to your new collection, you can add a `group_id` field to each vector. If you do this, Qdrant will assign each vector to its respective group. 
+
+Additionally, each vector can now be allocated to a shard. You can specify the `shard_key_selector` for each individual vector. In this example, you are upserting data belonging to `user_1` to the Canadian region.
+
+```python
+client.upsert(
+    collection_name="{collection_name}",
+    points=[
+        models.PointStruct(
+            id=1,
+            payload={"group_id": "user_1"},
+            vector=[0.9, 0.1, 0.1],
+            shard_key_selector="canada",
+        ),
+    ],
+)
+```
+Keep in mind that the data for each `group_id` is isolated. In the example below, `user_1` vectors are kept separate from `user_2`. The first user will be able to access their data in the Canadian portion of the cluster, while `user_2 `might only be able to retrieve USA-based information.
+
+```python
+client.upsert(
+    collection_name="{collection_name}",
+    points=[
+        models.PointStruct(
+            id=1,
+            payload={"group_id": "user_1"},
+            vector=[0.9, 0.1, 0.1],
+            shard_key_selector="canada",  
+        ),
+        models.PointStruct(
+            id=2,
+            payload={"group_id": "user_1"},
+            vector=[0.1, 0.9, 0.1],
+            shard_key_selector="canada", 
+        ),
+        models.PointStruct(
+            id=3,
+            payload={"group_id": "user_2"},
+            vector=[0.1, 0.1, 0.9],
+            shard_key_selector="usa", 
+        ),
+    ],
+)
+```
+
+## Retrieve data via filters
+
+The access control setup is completed as you specify the criteria for data retrieval. When searching for vectors, you need to use a `query_filter` along with `group_id` to filter vectors for each user. 
+
+```python
+client.search(
+    collection_name="{collection_name}",
+    query_filter=models.Filter(
+        must=[
+            models.FieldCondition(
+                key="group_id",
+                match=models.MatchValue(
+                    value="user_1",
+                ),
+            ),
+        ]
+    ),
+    query_vector=[0.1, 0.1, 0.9],
+    limit=10,
+)
+```
+
+## Performance considerations
+
+The speed of indexation may become a bottleneck if you are adding large amounts of data in this way, as each user's vector will be indexed into the same collection. To avoid this bottleneck, consider _bypassing the construction of a global vector index_ for the entire collection and building it only for individual groups instead.
+
+By adopting this strategy, Qdrant will index vectors for each user independently, significantly accelerating the process.
+
+To implement this approach, you should:
+
+1. Set `payload_m` in the HNSW configuration to a non-zero value, such as 16.
+2. Set `m` in hnsw config to 0. This will disable building global index for the whole collection.
```

Comment:
I could be wrong but I think this will slow down vector search. Or am I missing something?

Reply:
@timvisee What do you think? If this is the case, how would we address it?